### PR TITLE
fix(security): remediate 19 Dependabot and CodeQL alerts

### PR DESCRIPTION
**Pull request by djdanielsson**
## Summary
- Remediates all 16 open Dependabot vulnerability alerts and all 3 open CodeQL code scanning alerts (19 total)
- Upgrades vulnerable Python and frontend dependencies to patched versions
- Extracts path validation logic in Galaxy Proxy to resolve CodeQL path-injection findings

### Python dependency upgrades (uv.lock)
| Package | From | To | Severity | Issue |
|---------|------|----|----------|-------|
| idna | 3.11 | 3.15 | medium | CVE bypass in `idna.encode()` |
| urllib3 | 2.6.3 | 2.7.0 | high | Header forwarding across origins + decompression bomb bypass |
| python-multipart | 0.0.22 | 0.0.29 | high + medium | DoS via unbounded headers and preamble/epilogue |
| pip | 26.0.1 | 26.1.1 | medium | Untrusted control sphere + zip/tar confusion |
| pytest | 9.0.2 | 9.0.3 | medium | Vulnerable tmpdir handling |
| pygments | 2.19.2 | 2.20.0 | low | ReDoS in GUID matching |
| requests | 2.32.5 | 2.34.2 | medium | Insecure temp file reuse |

### Frontend dependency upgrades (package-lock.json)
| Package | From | To | Severity | Issue |
|---------|------|----|----------|-------|
| lodash | 4.17.23 | 4.18.1 | high + medium | Code injection via `_.template` + prototype pollution |
| postcss | 8.5.8 | 8.5.15 | medium | XSS via unescaped `</style>` in stringify |
| vite | 6.4.1 | 6.4.2 | high + medium | Path traversal in optimized deps + WebSocket file read |
| picomatch | 4.0.3 | 4.0.4 | medium | Method injection in POSIX classes + ReDoS |

### Code scanning fixes (galaxy_proxy/proxy/server.py)
- Extracted `_validate_tarball_dir()` helper function to isolate path sanitization from the endpoint handler, resolving 3 CodeQL `py/path-injection` alerts (alerts #11, #12, #14)
- Fixed `FileNotFoundError` handler: changed `break` to `continue` so symlink checks are not skipped for parent components when an intermediate path segment does not exist (addresses Copilot review feedback)
- Added 3 targeted tests: symlink leaf, symlink parent, and `..` traversal to cover the hardened path validation edge cases

## Test plan
- [x] `tox -e lint` passes
- [x] `tox -e unit` passes (2361 passed, 78 skipped)
- [x] `npm audit` reports 0 vulnerabilities
- [x] New tests cover symlink and path traversal edge cases
- [x] No docs changes needed (dependency-only + internal refactor)

### CODE REVIEW

File: src/galaxy_proxy/proxy/server.py [hunk 1]
```diff
@@ -474,6 +454,43 @@ async def convert_tarballs(tarball_dir: str) -> dict[str, list[str]]:
...
+    Raises:
+        HTTPException: On symlinks, disallowed roots, or non-directories.
+    """
+    raw = Path(tarball_dir)
+    for component in (raw, *raw.parents):
+        try:
+            if component.is_symlink():
+                raise HTTPException(status_code=400, detail="Symlinks not allowed")
+        except FileNotFoundError:
+            break
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply src/galaxy_proxy/proxy/server.py]: Fixed in 6b0fde4: changed `break` to `continue` in the `FileNotFoundError` handler so parent components are always checked for symlinks even when an intermediate path segment does not exist. Also added `test_convert_rejects_symlink_parent` which exercises the exact scenario described (symlink as a parent directory component).

File: src/galaxy_proxy/proxy/server.py [hunk 2]
```diff
@@ -474,6 +454,43 @@ async def convert_tarballs(tarball_dir: str) -> dict[str, list[str]]:
...
+
+    allowed_roots = (Path(tempfile.gettempdir()).resolve(), Path("/sessions").resolve())
+    if not any(resolved.is_relative_to(root) for root in allowed_roots):
+        raise HTTPException(
+            status_code=400,
+            detail="Path must be under a session or temp directory",
+        )
+
+    if not resolved.is_dir():
+        raise HTTPException(status_code=400, detail=f"Not a directory: {tarball_dir}")
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply src/galaxy_proxy/proxy/server.py:489]: Added in 6b0fde4: three new test cases in `TestConvertTarballs`:
- `test_convert_rejects_symlink_in_path` — symlink as the leaf directory
- `test_convert_rejects_symlink_parent` — symlink as a parent component (`link_parent/child`)
- `test_convert_rejects_dotdot_traversal` — path with `..` segments attempting root escape

All three pass in `tox -e unit` (2361 passed).
